eth/AccountManager.cpp

```diff
@@ -0,0 +1,230 @@
+/*
+	This file is part of cpp-ethereum.
+
+	cpp-ethereum is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	cpp-ethereum is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with cpp-ethereum.  If not, see <http://www.gnu.org/licenses/>.
+*/
+/** @file AccountManager.cpp
+ * @author Yann <yann@ethdev.com>
+ * @date 2016
+ */
+
+#include <libdevcore/SHA3.h>
+#include <libdevcore/FileSystem.h>
+#include <libethcore/ICAP.h>
+#include <libethcore/KeyManager.h>
+#include "AccountManager.h"
+using namespace std;
+using namespace dev;
+using namespace dev::eth;
+
+bool AccountManager::execute(int argc, char** argv)
+{
+	if ((string)argv[1] == "wallet")
+	{
+		if (4 < argc && (string)argv[2] == "import")
+		{
+			keyManager();
+			string file =  argv[3];
+			string name = argv[4];
+			std::string pw;
+			KeyPair k = m_keyManager->presaleSecret(contentsString(file), [&](bool){ return (pw = getPassword("Enter the passphrase for the presale key: ")); });
+			m_keyManager->import(k.secret(), name, pw, "Same passphrase as used for presale key");
+		}
+		else
+			cout
+				<< "	wallet import <file> <name>	Import a presale wallet into a key with the given name." << endl;
+
+		m_keyManager.reset();
+		return true;
+	}
+	else if ((string)argv[1] == "account")
+	{
+		if (2 < argc && (string)argv[2] == "list")
+		{
+			keyManager();
+			if (m_keyManager->store().keys().empty())
+				cout << "No keys found." << endl;
+			else
+			{
+				vector<u128> bare;
+				AddressHash got;
+				for (auto const& u: m_keyManager->store().keys())
```

Comment:
Would be great if this routine could work without a key manager, i.e. without master passphrase.